src/__test__/simulator/sdk.ts

```diff
@@ -218,9 +198,10 @@ class sdk2 {
 		}
 		let p2: P2Signature | null = null;
 		let round = 1;
+		let signUnSub: any;
```

Comment:
I noticed we use any for `signUnSub`. We can improve this in the PR. We can do something like 
```typescript
const signUnSub = await new Promise<Unsubscribe>((resolve) => {
   const signUnSubTemp = onSnapshot(...) => {
      .
      .
      .
      resolve(signUnSubTemp);
   } 
}
signUnSub();
```

Reply:
Also why signUnSub and keygenUnSub is called in a different way. keygenUnSub is called outside the promise but signUnSub is called inside the promise, if possible can we unify both the approach.